library/ssl_srv.c

```diff
@@ -1785,7 +1785,7 @@ static int ssl_parse_client_hello( mbedtls_ssl_context *ssl )
                                             "during renegotiation" ) );
                 mbedtls_ssl_send_alert_message( ssl, MBEDTLS_SSL_ALERT_LEVEL_FATAL,
                                                 MBEDTLS_SSL_ALERT_MSG_HANDSHAKE_FAILURE );
-                return( MBEDTLS_ERR_SSL_BAD_HS_CLIENT_HELLO );
+                return( MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE );
```

Comment:
UNEXPECTED_MESSAGE seems more appropriate?

Reply:
My understanding of UNEXPECTED_MESSAGE was that it's returned when we receive a record or handshake message of the wrong type, not when the content of a message isn't what we expected.

On second thought, it seems that `MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER` is what we should use here, because the message is semantically wrong in the context of renegotiation.